src/interfaces.ts

```diff
@@ -200,6 +200,7 @@ export interface CompilerPlugin {
     afterProgramValidate?: (program: Program) => void;
     beforeProgramTranspile?: (program: Program, entries: TranspileObj[], editor: AstEditor) => void;
     afterProgramTranspile?: (program: Program, entries: TranspileObj[], editor: AstEditor) => void;
+    beforeProgramDispose?: (program: Program) => void;
```

Comment:
Can you update this to follow the newer `PluginHandler` pattern? I know some of our other events have ordered params, but we've been moving all new events to this new pattern to make it more flexible.

```
beforeProgramDispose?: PluginHandler<BeforeProgramDisposeEvent>;
```

```
interface BeforeProgramDisposeEvent {
    program: Program;
}

Reply:
Good idea!